ykcompile/src/lib.rs

```diff
@@ -181,20 +240,50 @@ impl TraceCompiler {
         local: Local,
         constant: &ConstantInt,
     ) -> Result<(), CompileError> {
-        let reg = self.local_to_reg(local)?;
+        let loc = self.local_to_location(local)?;
         let c_val = constant.i64_cast();
-        dynasm!(self.asm
-            ; mov Rq(reg), QWORD c_val
-        );
+        match loc {
+            Location::Register(reg) => {
+                dynasm!(self.asm
+                    ; mov Rq(reg), QWORD c_val
+                );
+            }
+            Location::Stack(offset) => {
+                if c_val <= u32::MAX.into() {
+                    let val = c_val as u32 as i32;
+                    dynasm!(self.asm
+                        ; mov QWORD [rsp+offset], val
+                    );
+                } else {
```

Comment:
Let's comment this up so that we remember what it does a few months down the line?

Reply:
This is being changed in the last commit. But I can add a comment to that commit to explain what it does.

Reply:
Fixed in a60544c